proposals/044-integration-model.md

```diff
@@ -0,0 +1,203 @@
+Running integrations can be suspended and resumed without destroying the deployment.
+
+
+![Integration state model](images/integration-state.png "Integration State Model")
+
+| State       | Description |
+| ----------- |------------ |
+| Draft | Initial state of an integration. The integration is not yet deployed |
+| Active | Integration is deployed and running |
+| Inactive | Integration is deployed but is not running|
+| Undeployed | An integration has been undeployed | 
+| Error | The integration is deployed but in an error state |
+
+### Draft
+
+Right after an integration is created it enters the _Draft_ state. This means 
+the integration is not yet deployed. An integration leaves the draft state either by being published or deleted (in which case the whole integration is removed)
+
+After an activation there is no way back to the _Draft_ state.
+
+### Active
+
+An _active_ integration is a running integration. This state can be either created from a _Draft_ state when the integration gets published of from an _Inactive_ state when a suspended integration is started again.
+
+### Inactive
+
+An _inactive_ integration is an integration which was previously active and has been suspended by some action. An _inactive_ integration can be resumed or undeployed
+
+### Undeployed
+
+An integration which was published is never deleted but can be undeployed. An undeployed integration can be re-deployed (and activated) if the user chose the version of this integration to be reactivated.
+
+### Error
+
+The error state indicates that the integration is not running. This state can be entered consciously but only as a side effect. An integration can enter the error state any time. 
+
+## State reconciliation 
+
+Integrations follow a declarative reconciliation paradigm. An integration is always in a _currentState_. As consequence of an _action_ (see below) a new _targeState_ can be set. It is now the duty of the Syndesis backend to reconcile the _currentState_ to become the _targetState_.
+This modelled is closely to the Kubernetes state model itself.
+
+Reconciliation itself can be in one of two states:
+
+* _Pending_ when `currentState != targetState`
+* _Ok_ when `currentState == targetState`
+
+This state is modelled implicitely by looking at both state variables. 
+
+`Error` can never be a `targetState` but it can be a `currentState`. The backend controllers try to get out of the error state but might give up at some point. 
+
+Every state change (current or target) should be tracked. 
+
+## Operations
+
+The following operations change the `targetState`. The `targetState` could be updated by simply changing the Integration's field, but it is recommended to use a more semantic rich REST API call for perfoming this actions (i.e. having specific endpoint which can be used by the UI to trigger the targetState change).
+
+* **Publish** sets the target state to _Active_. This operation can be called from the states:
```

Comment:
If you save but not "publish" nothing will happen on the running integration.

When changing an integration the following steps and checks are performed. 
Please keep in mind that all integration objects with the same name belong together, they are just variants in different versions.

* If there is already an integration with this name in state _Draft_, update this integration.
* If there is no integration in _Draft_ but one in _Active_, the active integration is cloned to a new integration with the same name and in state _Draft_. This integration then can be changed freely and saved to the DB.
* When the changed integration gets published, it's `targetState` is set from "Draft" to "Active". 
* A controller in syndesis-rest detects, that integration with `currentState == Draft` and `targetStat == Active`. The controller then performs these operations in the background:
  - For an integration with `currentState == Active|Inactive` set `targetState = Undeployed`
  - Create / Update the GitHub repo
  - Trigger an OpenShift build  

(i added this to the document)